src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,35 @@
+package duke.command;
+
+import duke.Ui;
+import duke.TaskList;
+import duke.exceptions.DukeException;
+
+public abstract class Command {
+
+    protected Ui ui;
+    protected TaskList tasks;
+    private int targetIndex = -1;
```

Comment:
This private variable is declared in this abstract class but not used. Do you need this variable?